geary-core/src/main/kotlin/com/mineinabyss/geary/ecs/api/entities/GearyEntity.kt

```diff
@@ -53,6 +53,10 @@ public value class GearyEntity(public val id: GearyEntityId) {
 
     public inline fun <reified T : GearyComponent, reified C : GearyComponent> getRelation(): T? = getRelation(C::class)
 
+    public inline fun <reified T : GearyComponent> getRelation(componentKlass: KClass<GearyComponent>): T? {
```

Comment:
This is necessary because right now

getRelation<T,C>() delegates to getRelation<T>(C:class) which delegates to fancy version of getRelation(T::class, C::class::class)